content/cucumber/browser-automation.md

```diff
@@ -16,86 +17,149 @@ Automation tools such as:
+  element.submit
+ end
+
+ Then("^the page title should start with "([^"]*)"$") do
+  wait = Selenium::WebDriver::Wait.new(:timeout => 10)
+  wait.until { driver.title.downcase.start_with? "cheese!" }
+  puts "Page title is #{driver.title}"
+   browser.close
+ end
+```
+
+## Watir Webdriver
 
-TODO
+Watir, pronounced water, is an open-source (BSD) family of Ruby libraries for automating web browsers. It allows you to write tests that are easy to read and maintain. It is simple and flexible.
+
+Watir drives browsers the same way people do. It clicks links, fills in forms, presses buttons. Watir also checks results, such as whether expected text appears on the page.
+
+Watir is a family of Ruby libraries but it supports your app no matter what technology it is developed in. While Watir supports only Internet Explorer on Windows, Here comes Watir-WebDriver to solve single browser testing and support Chrome, Firefox, Internet Explorer, Opera and also running in headless mode (HTMLUnit).
+
+Now without wasting anytime quickly jump in to a sample UI testing program using Watir
+
+ ```ruby
+ require "rubygems"
+ require "rspec"
+ require "watir-webdriver"
+
+ describe "google.com" do
+   let(:browser) { @browser ||= Watir::Browser.new :firefox }
+   before { browser.goto "http://google.com" }
+   browser.text_field(:name => "q").set "watir"
+   browser.button.click
+   browser.div(:id => "resultStats").wait_until_present
+   browser.title.should == "watir - Google Search"
+   after { browser.close }
+ end
+```
+
+Now let us incorporate Cucumber to this simple test
+
+```gherkin
+ Feature: Search In order to use Google users must be able to search for content
+   Scenario: Search for a term
+     Given I have entered "watir" into the query
+     When I click "search"
+     Then I should see some results
+```
+
+```ruby
+require "watir-webdriver"
+require "rspec/expectations"
+
+Given /^I have entered "([^"]*)" into the query$/ do |term|
+ @browser ||= Watir::Browser.new :firefox
+ @browser.goto "google.com"
+ @browser.text_field(:name => "q").set term
+end
+
+When /^I click "([^"]*)"$/ do |button_name|
+ @browser.button.click
+end
+
+Then /^I should see some results$/ do
+ @browser.div(:id => "resultStats").wait_until_present
+ @browser.div(:id => "resultStats").should exist
+ @browser.close
+end
+```
 
 ## Serenity BDD
 
 Serenity BDD is an open source reporting library that helps you write better
 structured, more maintainable automated acceptance criteria. Serenity also produces
 rich meaningful test reports (or "living documentation") that report not only the
-test results, but also which Features have been tested. 
+test results, but also which Features have been tested.
 
 A detailed tutorial on using Cucumber-JVM with Serenity can be found
 [here](http://thucydides.info/docs/articles/an-introduction-to-serenity-bdd-with-cucumber.html).
 
 The above Scenario's Step Definitions might be written for Serenity like this:
 
-[carousel]
-
 ```ruby
```

Comment:
'ruby' is mentioned twice (with comment about Java only)?
Also tag '[carousel]' is missing?

Reply:
We no longer need carousel, that is a left over from the previous site setup.

Reply:
Yeah, I figured that out. removed the comment about carousel :)